packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -123,7 +137,7 @@ export class Tracing implements InstrumentationListener, SnapshotterDelegate, Ha
     const state = this._state;
     const suffix = state.filesCount ? `-${state.filesCount}` : ``;
     state.filesCount++;
-    state.traceFile = path.join(this._tracesDir, `${state.traceName}${suffix}.trace`);
+    state.traceFile = path.join(this._tracesDir!, `${state.traceName}${suffix}.trace`);
```

Comment:
nit: you can put non-nullable tracesDir into state to avoid all question marks.

Reply:
done